From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 4 Dec 2011 15:26:23 +0100
Subject: remote_driver: don't fail if keepalive check fails

Otherwise connections to older libvirt abort with:

$ virsh -c qemu+ssh://host.example.com/system list
error: invalid connection pointer in virDrvSupportsFeature
error: failed to connect to the hypervisor

Tested against 0.8.3 and 0.9.8-rc2.
---
 src/remote/remote_driver.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/src/remote/remote_driver.c b/src/remote/remote_driver.c
index 556c90c..1fb0eca 100644
--- a/src/remote/remote_driver.c
+++ b/src/remote/remote_driver.c
@@ -678,10 +678,8 @@ doRemoteOpen (virConnectPtr conn,
         rc = call(conn, priv, 0, REMOTE_PROC_SUPPORTS_FEATURE,
                   (xdrproc_t)xdr_remote_supports_feature_args, (char *) &args,
                   (xdrproc_t)xdr_remote_supports_feature_ret, (char *) &ret);
-        if (rc == -1)
-            goto failed;
 
-        if (ret.supported) {
+        if (rc != -1 && ret.supported) {
             priv->serverKeepAlive = true;
         } else {
             VIR_INFO("Disabling keepalive protocol since it is not supported"
